pkg/ingester/flush.go

```diff
@@ -241,19 +243,19 @@ func (i *Ingester) flushUserSeries(flushQueueIndex int, userID string, fp model.
 
 	userState, ok := i.userStates.get(userID)
 	if !ok {
-		return nil
+		return "NoUser", nil
 	}
 
 	series, ok := userState.fpToSeries.get(fp)
 	if !ok {
-		return nil
+		return "NoSeries", nil
 	}
 
 	userState.fpLocker.Lock(fp)
 	reason := i.shouldFlushSeries(series, fp, immediate)
 	if reason == noFlush {
 		userState.fpLocker.Unlock(fp)
-		return nil
+		return reason.String(), nil
```

Comment:
This feels odd - calling a method on a constant value.

Reply:
`noFlush.String()` didn't work for me, not sure why. But I will rework it to return `flushReason` instead, as you suggested elsewhere.